### PR TITLE
[BEAM-5151][SQL] create external table

### PR DESCRIPTION
**Pull request by amaliujia**
Use `CREATE EXTERNAL TABLE` to replace `CREATE TABLE`

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by akedin** [comment]
I'm fine with this approach. Don't we have any tests for this though? We should if we don't.

**Comment by amaliujia** [comment]
@akedin Just want to collect some option first. If people think it is fine, I will continue working on this PR.

**Comment by amaliujia** [comment]
updated tests to reflect this change.

**Comment by amaliujia** [comment]
run java postcommit

**Comment by amaliujia** [comment]
run java postcommit

**Comment by amaliujia** [comment]
java run precommit

**Comment by amaliujia** [comment]
run java precommit

**Comment by amaliujia** [comment]
run java postcommit

**Comment by amaliujia** [comment]
run java precommit

**Comment by akedin** [comment]
Second thought, we should throw an exception for `CREATE TABLE` saying something like "CREATE TABLE is not supported in Beam SQL. Please use CREATE EXTERNAL TABLE instead."

File: sdks/java/extensions/sql/src/main/codegen/includes/parserImpls.ftl [hunk 1]
```diff
@@ -191,6 +191,18 @@ SqlCreate SqlCreateExternalTable() :
...
 
+SqlCreate SqlCreateTableNotSupportedMessage(Span s, boolean replace) :
+{
+
+}
+{
+  <TABLE>
+
+  {
+    return new SqlCreateTableNotSupportedMessage(s.end(this), replace);
```
Comment (akedin) [inline sdks/java/extensions/sql/src/main/codegen/includes/parserImpls.ftl]: is it possible to throw from here directly without creating a new class?

Comment (amaliujia) [reply sdks/java/extensions/sql/src/main/codegen/includes/parserImpls.ftl]: Yes it is. I changed the code to throw the exception and not create a new class.

**Comment by amaliujia** [comment]
image

**Comment by amaliujia** [comment]
run java postcommit

**Comment by amaliujia** [comment]
run java postcommit

**Comment by amaliujia** [comment]
Ping @XuMingmin @xumingming just in case this PR will have side effect on your side.

**Comment by XuMingmin** [comment]
@amaliujia I'm ok with the change given `CREATE EXTERNAL TABLE` is more accurate. Will handle the impact in our side when upgrading to next release version.

**Pull request merged**